File: backend/app/schemas/trade_review_v2.py
"""Pydantic schemas for Trade Review Engine V2 — deterministic structured reviews."""

from typing import Literal, Optional

from pydantic import BaseModel, Field


ScoreLabel = Literal["excellent", "good", "average", "weak", "critical"]
MistakeSeverity = Literal["info", "warning", "critical"]
MistakeCategory = Literal["setup", "entry", "exit", "risk", "psychology", "process"]


class ReviewDimensionScore(BaseModel):
    dimension: str
    score: int = Field(ge=0, le=100)
    label: ScoreLabel
    reason: str
    evidence: list[str] = Field(default_factory=list)
    improvement: Optional[str] = None


class MistakeTag(BaseModel):
    tag: str
    severity: MistakeSeverity
    category: MistakeCategory
    explanation: str
    suggested_fix: str


class TradeReviewV2Response(BaseModel):
    trade_id: int
    symbol: str
    setup: Optional[str] = None
    direction: str
    status: str
    reviewed_at: str
    overall_score: int = Field(ge=0, le=100)
    overall_label: ScoreLabel
    verdict: str
    dimension_scores: list[ReviewDimensionScore] = Field(default_factory=list)
    mistake_tags: list[MistakeTag] = Field(default_factory=list)
    strengths: list[str] = Field(default_factory=list)
    weaknesses: list[str] = Field(default_factory=list)
    what_should_have_happened: list[str] = Field(default_factory=list)
    next_time_rules: list[str] = Field(default_factory=list)
    review_questions: list[str] = Field(default_factory=list)
    evidence: dict = Field(default_factory=dict)
    source: str = "deterministic_v2"


class TradeReviewBatchSummary(BaseModel):
    avg_score: float
    common_mistakes: list[str] = Field(default_factory=list)
    strongest_dimension: Optional[str] = None
    weakest_dimension: Optional[str] = None


class TradeReviewBatchResponse(BaseModel):
    generated_at: str
    count: int
    reviews: list[TradeReviewV2Response] = Field(default_factory=list)
    summary: TradeReviewBatchSummary

File: frontend/src/types/tradeReviewV2.ts
export type ScoreLabel = 'excellent' | 'good' | 'average' | 'weak' | 'critical'
export type MistakeSeverity = 'info' | 'warning' | 'critical'
export type MistakeCategory = 'setup' | 'entry' | 'exit' | 'risk' | 'psychology' | 'process'

export interface ReviewDimensionScore {
  dimension: string
  score: number
  label: ScoreLabel
  reason: string
  evidence: string[]
  improvement: string | null
}

export interface MistakeTag {
  tag: string
  severity: MistakeSeverity
  category: MistakeCategory
  explanation: string
  suggested_fix: string
}

export interface TradeReviewV2Response {
  trade_id: number
  symbol: string
  setup: string | null
  direction: string
  status: string
  reviewed_at: string
  overall_score: number
  overall_label: ScoreLabel
  verdict: string
  dimension_scores: ReviewDimensionScore[]
  mistake_tags: MistakeTag[]
  strengths: string[]
  weaknesses: string[]
  what_should_have_happened: string[]
  next_time_rules: string[]
  review_questions: string[]
  evidence: Record<string, unknown>
  source: string
}

export interface TradeReviewBatchSummary {
  avg_score: number
  common_mistakes: string[]
  strongest_dimension: string | null
  weakest_dimension: string | null
}

export interface TradeReviewBatchResponse {
  generated_at: string
  count: number
  reviews: TradeReviewV2Response[]
  summary: TradeReviewBatchSummary
}
